ui/src/app/pipes/unit-conversion.pipe.ts

```diff
@@ -0,0 +1,25 @@
+import { Pipe, PipeTransform } from '@angular/core'
+
+@Pipe({
+  name: 'durationToSeconds',
+})
+export class DurationToSecondsPipe implements PipeTransform {
+  transform (duration: string): number {
+    if (!duration) return 0
+    const unit = units.filter(u => duration.includes(u)).reduce((a, b) => a.length > b.length ? a : b )
```

Comment:
I hope I'm not the only one that thinks this is crazy. I mean yes it will work but its wild.
Normally I would just use a regex match to split off the units, and match on them.
`^([0-9]*(\.[0-9]+)?)(ns|µs|ms|s|m|d)$` - your number is group 1, and your units are group 3.